projects/vdk-core/src/vdk/internal/builtin_plugins/ingestion/ingester_router.py

```diff
@@ -143,7 +143,8 @@ def __ingest_object(
                 payload, destination_table, method, target, collection_id
             )
         except Exception as e:
-            self._log.warn("Failed to send object for ingestion." f"Exception was: {e}")
+            errors.report(ResolvableBy.USER_ERROR, e)
```

Comment:
Why not just call report_and_rethrow, it contains the exact same two lines.

Reply:
I experimented with both and with report_and_rethrow the stack trace is more confusing. 

See the comparison https://www.diffchecker.com/U4Sgb28h/ 

The 2 newly added method calls sort of branch out . Re-raising the error keeps the stack trace more linear hence easier to understand. 

If we simply want to report the error and allow it to continue propagating as if it had not been caught at all (keeping the stack trace clean), we should use raise by itself. 

That means likely we should discourage the use of report_and_throw in favour of report() and raise